Results Viewer Improvements

anandthakker opened the pull request:
- Accept multiple baseurl= query params, and show the results side by side
 - Start without sorting by metrics, and allow for resetting to that

@drewbo lots of changes here, so probably a bit hard to review the code fully.  You can try it out locally with:
 - `npm run start-viewer`
 - go to http://localhost:9966/view.html?baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.ts-9-500/snapshots/segnet_iter_20000.results/&baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.ts-9-1000/snapshots/segnet_iter_20000.results/&baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.ts-9-2000/snapshots/segnet_iter_20000.results/&baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.ts-9-4000/snapshots/segnet_iter_20000.results/

Commit:
Results Viewer Improvements

 - Accept multiple baseurl= query params, and show the results side by side
 - Start without sorting by metrics, and allow for resetting to that

drewbo commented on results-viewer/view.js:
```diff
@@ -90,47 +103,59 @@ const view = (params, state, send) => {
     <button onclick=${() => send('app:sort', { key: 'correctness_score:ascending' })}>Least Correct</button>
     <button onclick=${() => send('app:sort', { key: 'completeness_score:descending' })}>Most Complete</button>
     <button onclick=${() => send('app:sort', { key: 'completeness_score:ascending' })}>Least Complete</button>
+    <button onclick=${() => send('app:sort', { key: 'completeness_score:ascending' })}>Least Complete</button>
```
Is this button duplicated?

anandthakker replied:
Yup -- about to push a quick style tweak, I'll remove this as well.

Commit:
Tweak css, drop duplicate button

anandthakker commented:
@drewbo fixed.  Also, this is a nice one: http://localhost:9966/view.html?baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.phi-3.1/snapshots/segnet_iter_10000.results/&baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.phi-3.1/snapshots/segnet_iter_20000.results/&baseurl=http://ds-skynet.s3.amazonaws.com/models/segnet.phi-3.1/snapshots/segnet_iter_30000.results/#1/0/0

anandthakker commented:
(we hit 30K on phi)

drewbo commented on results-viewer/view.js:
```diff
@@ -225,3 +252,37 @@ function logged (view, tag) {
   }
 }
 
+function stripCommon (strings) {
```
Can you add a comment explaining this function?

drewbo commented on results-viewer/view.js:
```diff
@@ -90,47 +103,58 @@ const view = (params, state, send) => {
           var tile = getTile(item)
           var image = tile ? getSatelliteTileURL(query, tile[0], tile[1], tile[2])
-            : (baseurl + item.input)
+            : (baseurls[0] + item.input)
```
Is this so that it matches the ground truth of the first baseurl set?

anandthakker replied:
Oh yeah, so this is here for cases where the input data is not a standard map tile (so that we can't just use getSatelliteTileURL to construct a url based on x/y/z coords).  In that case, `baseurls[0] + item.input` would grab the "xxx_input.png" image that sits alongside "xxx_groundtruth.png" and "xxx_prediction.png".

The only case where we've used the latter is for the DG SpaceNet thing, where the images weren't tiled.

(Note that, as of https://github.com/developmentseed/skynet-internal/issues/3, those `xxx_input.png` images wouldn't be accessible on our ds-skynet working bucket; we'd need copy them to them to skynet-results to view.)

drewbo commented:
Getting CORS errors on the s3 pngs

drewbo committed:
Add missing tilebelt dependency

anandthakker commented:
> Getting CORS errors on the s3 pngs

Is this only when you click on them to show on the map?

Commit:
Add comment for stripCommon

Commit:
Misc fixes

 - Better handling of digits in stripCommon (see code comment)
 - Add trailing slash to baseurls
 - Use '//' instead of 'http://' for tile URLs

Commit:
Fixup column styling

drewbo commented:
> > Getting CORS errors on the s3 pngs

> Is this only when you click on them to show on the map?

Yes
